spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/PulsarConsumerTestUtil.java

```diff
@@ -67,6 +69,13 @@ public SchemaSpec<T> fromTopic(String topic) {
 		return this;
 	}
 
+	@Override
+	public ConditionsSpec<T> withSchema(Schema<T> schema) {
+		Assert.notNull(schema, "Schema must not be null");
```

Comment:
Moved this up in order-must-be-executed for a nice readable mental flow and also added null check. Did you leave it out for a particular reason that I am missing?